types/web3/index.d.ts

```diff
@@ -47,7 +47,7 @@ declare class Web3 {
     shh: Shh;
     givenProvider: Provider;
     providers: Providers;
-    setProvider(provider: Provider): void;
+    setProvider(provider: Provider): boolean;
```

Comment:
Okay, the docs say it returns a `boolean`. [But what does this boolean tell me?](https://web3js.readthedocs.io/en/1.0/include_package-core.html?highlight=setProvider#returns) I think this is a mistake in the docs, it will either throw something or return true. So a function that always returns `true` returns `void` imo. I would prefer to leave it as returning a `void` and maybe add a comment why we differ here from the documentation (which makes no sense at all for this).